### ISSUE TITLE
RGBA values produce different result for imshow and for markers

### ISSUE THREAD

**Issue by astrofrog** [matplotlib/matplotlib#7201]
In the following example, I produce 9 colors ranging from completely transparent red to completely opaque red. In the case of imshow, the lighter colors actually end up being closer to gray, whereas for the scatter markers, the colors look as one would expect. What could be causing this?

```python
import numpy as np
import matplotlib.pyplot as plt

data = np.arange(9, dtype=float).reshape((3, 3)) / 8.

color = np.array([1, 0, 0, 1], dtype=float)

array = color * np.ones(data.shape + (4,))
array[:, :, 3] *= data

fig = plt.figure()
ax = fig.add_subplot(1, 1, 1)
ax.imshow(array, vmin=0, vmax=1, interpolation='nearest')
for i in range(3):
    for j in range(3):
        ax.scatter(i, j, s=500, facecolor='white', edgecolor='white', alpha=1.0)
        ax.scatter(i, j, s=400, facecolor=array[j, i], edgecolor='white')
fig.savefig('colors.png')
```

![colors](https://cloud.githubusercontent.com/assets/314716/18963537/4f59c48e-866d-11e6-8c60-d66aeb1d0cdd.png)

**Comment by QuLogic** [comment]
This works in `2.x` as you'd expect:

![colors](https://cloud.githubusercontent.com/assets/302469/18971994/55733966-8665-11e6-98ba-7c2ebe513c2a.png)

**Comment by QuLogic** [comment]
Maybe a dupe of #5949, but it's fixed by #5718.

**Comment by QuLogic** [comment]
Sorry, that reference should be to #3343.
